src/psyclone/tests/psyir/tools/call_tree_utils_test.py

```diff
@@ -441,3 +448,90 @@ def test_call_tree_utils_const_argument():
     # Make sure the constant '0' is not listed
     assert "0" not in read_write_info.signatures_read
     assert Signature("0") not in read_write_info.signatures_read
+
+
+# -----------------------------------------------------------------------------
+@pytest.mark.usefixtures("clear_module_manager_instance")
+def testcall_tree_utils_non_local_inout_parameters(capsys):
+    '''Tests the collection of non-local input and output parameters.
+    '''
+    Config.get().api = "dynamo0.3"
+    ctu = CallTreeUtils()
+
+    test_file = os.path.join("driver_creation", "module_with_builtin_mod.f90")
+    psyir, _ = get_invoke(test_file, "dynamo0.3", 0, dist_mem=False)
+    schedule = psyir.invokes.invoke_list[0].schedule
+
+    test_dir = os.path.join(get_base_path("dynamo0.3"), "driver_creation")
+    mod_man = ModuleManager.get()
+    mod_man.add_search_path(test_dir)
+
+    # The example does contain an unknown subroutine (by design), and the
+    # infrastructure directory has not been added, so constants_mod cannot
+    # be found:
+    rw_info = ctu.get_in_out_parameters(schedule,
+                                        collect_non_local_symbols=True)
+    out, _ = capsys.readouterr()
+    assert "Unknown routine 'unknown_subroutine - ignored." in out
+    assert ("Cannot find module 'constants_mod' - ignoring unknown symbol "
```

Comment:
It might be useful to include in this message the include/search path used when searching for 'constants_mod'.

Reply:
Not in this message - it's about a subroutine call which we don't know where it is coming from, i.e. it's not from a module. So we don't even use the paths here.

But there was another error message which should indeed list the paths, and I've added them.